ui/src/utils/rules.js

```diff
@@ -18,8 +18,15 @@ const ruleMatchesChannel = (rule, channel) => {
   // if a scheduled change does not exist at all
   // we never want this to match, otherwise all rules
   // without scheduled changes will always match any filter
+  // if a scheduled change is a deletion
+  // we never want scheduled deletions to match any filter
+  // because the scheduled change channel on deletions is null
+  // and the rule channel is a truthy value that might not equal 'channel'
   const scChannelMatches = rule.scheduledChange
-    ? rule.scheduledChange.channel === null ||
+    ? (rule.channel &&
+        ruleChannelMatches &&
+        rule.scheduledChange.channel === null) ||
+      (!rule.channel && rule.scheduledChange.channel === null) ||
```

Comment:
This appears to work, but I'm suspicious that something subtle is off here. I was expecting `rule.scheduledChange.change_type === 'delete'` to be treated specially - not to have _all_ scheduled change types have this logic applied. For `change_type` === `update` or `insert` the channel should be matching if either the `rule` or `rule.scheduledChange` channel matches.

Could you try to make this a bit more clear? Feel free to drop the `const` on `scChannelMatches` and write this as if/else statements if it helps.

Reply:
I wasn't sure whether to query for the change type initially or just make the function work for all scheduled changes. I appreciate the feedback and I'll work on making it clear.